docs/architecture/deep-dives/passkeys/implementations/relying-party/prf.md

```diff
@@ -0,0 +1,116 @@
+
+:::tip Client extensions vs. authenticator extensions
+
+The `prf` extension is defined above as a "client extension". This is to differentiate it from an
+"authenticator extension".
+
+The [distinction](https://www.w3.org/TR/webauthn-3/#sctn-extensions) between client extensions and
+authenticator extensions in WebAuthn lies in where they are processed and what they influence. An
+authenticator extension involves communication with and processing in both the client and
+authenticator, while a client extension involves only the client.
+
+The `prf` client extension _can_ be built on top of the the `hmac-secret` authenticator extension,
+but that is not a requirement.
+
+:::
+
+## How does Bitwarden use the PRF key?
+
+### During passkey registration
+
+When a passkey is registered for use with Bitwarden, we interrogate the attestation response to see
+if PRF is supported. If so, we allow the user to choose to enable encryption.
+
+![Registering a passkey with PRF](image.png)
+
+If the user chooses to enable encryption, we request an assertion from the authenticator, which will
+provide the key. (This is why you may be prompted multiple times to provide user verification when
+registering a PRF-capable passkey.)
+
+The 32-byte key returned from the authenticator is then stretched to a 64-byte two-part key using
+HKDF. The first 32 bytes will serve as the AES-256 encryption key and the second 32 bytes will serve
+as a message authentication code (MAC).
+
+Once we have this encryption key, we can encrypt the user's symmetric key as follows:
+
+    1. Create an RSA public/private key pair, specific to this credential.
+    2. Encrypt the RSA private key with the encryption key, producing `EncryptionKey(PrivateKey)`
+    3. Encrypt the user's symmetric key with the RSA public key, producing `PublicKey(UserSymmetricKey)`
+    4. Encrypt the RSA public key with the user's symmetric key, producing `UserSymmetricKey(PublicKey)`
+
+`EncryptionKey(PrivateKey)`, `PublicKey(UserSymmetricKey)`, and `UserSymmetricKey(PublicKey)` are
+persisted on the server and associated with the registered passkey. Note that the PRF key does
+**not** leave the client at any point, to ensure zero-knowledge encryption.
+
+:::tip Why an RSA key pair?
+
+Put simply: key rotation. In order to support
+[encryption key rotation](https://bitwarden.com/help/account-encryption-key/#rotate-your-encryption-key),
+a user must be able to re-encrypt the symmetric key for all registered passkeys. If the PRF key was
+used to directly encrypt the user's symmetric key then this operation would require gaining access
+to the PRF keys for each registered credential. The use of a public/private key pair allows the user
+to re-encrypt the new symmetric key with the public key, with the assurance that only the holder of
+the private key can decrypt it, and without having to provide access to the PRF key for each
+registered credential.
+
+:::
+
+### During passkey log in
```

Comment:
```suggestion
### During passkey login
```

Reply:
I literally just changed this from `login` -> `log in` :) I think I have a hard time making the right choice due the fact that I'm not a native speaker, could you maybe expand on why `log in` doesn't work here?

Reply:
This is where I was asked to the opposite change https://github.com/bitwarden/contributing-docs/pull/290#discussion_r1494640051 @trmartin4 

Also I think the feature is called "Log in with passkey", if that has any impact

Reply:
I can certainly sympathize, and this is one of the quirkier English adaptations of intent. Some scenarios to help with context:

1. I am a user and arrive at a screen for credentials. I am offered to "_log in_ with passkey" because I am taking that _action_. It's a use of the verb.
2. I am a documentation writer and see a section about a way to authenticate. I am documenting "_login_ with passkey" because I am working on a _concept_. It's a use of the noun.